### PR TITLE
Bugfix/phpstan level5

### PR DESCRIPTION
**Pull request by dereuromark**
After adding all missing docblocks, a lot of more issues appeared
the total count of errors was 360+, I managed to get them down to 251 (see baseline)

This also fixes (documented) return types to the actual ones.
Especially around nullable.

//EDIT: 207 issues now left in baseline

### CODE REVIEW

File: src/Propel/Generator/Behavior/I18n/I18nBehavior.php [hunk 1]
```diff
@@ -119,6 +142,9 @@ public function replaceTokens($string)
         ]);
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\I18n\I18nBehavior|\Propel\Generator\Behavior\I18n\I18nBehaviorObjectBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/I18n/I18nBehavior.php:146]:
```suggestion
     * @return \Propel\Generator\Behavior\I18n\I18nBehaviorObjectBuilderModifier
```

File: src/Propel/Generator/Behavior/I18n/I18nBehavior.php [hunk 2]
```diff
@@ -128,6 +154,9 @@ public function getObjectBuilderModifier()
         return $this->objectBuilderModifier;
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\I18n\I18nBehavior|\Propel\Generator\Behavior\I18n\I18nBehaviorQueryBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/I18n/I18nBehavior.php:158]:
```suggestion
     * @return \Propel\Generator\Behavior\I18n\I18nBehaviorQueryBuilderModifier
```

File: src/Propel/Generator/Behavior/I18n/I18nBehavior.php [hunk 3]
```diff
@@ -232,8 +279,10 @@ protected function addLocaleColumnToI18n()
 
     /**
      * Moves i18n columns from the main table to the i18n table
+     *
+     * @return void
      */
-    protected function moveI18nColumns()
+    protected function moveI18nColumns(): void
```
Comment (demkos) [inline src/Propel/Generator/Behavior/I18n/I18nBehavior.php:285]: Revert?

File: src/Propel/Runtime/ActiveRecord/NestedSetRecursiveIterator.php [hunk 4]
```diff
@@ -80,15 +95,19 @@ public function next()
             }
             $this->curNode = $nextNode;
         }
-
-        return $this->curNode;
```
Comment (gechetspr) [inline src/Propel/Runtime/ActiveRecord/NestedSetRecursiveIterator.php:84]: I'm a little bit confused by this one. Why we removed this?

File: src/Propel/Generator/Behavior/I18n/I18nBehaviorObjectBuilderModifier.php [hunk 5]
```diff
@@ -156,8 +217,13 @@ protected function addGetCurrentTranslation()
         ]);
     }
 
-    // FIXME: the connection used by getCurrentTranslation in the generated code
-    // cannot be specified by the user
+    /**
+     * FIXME: the connection used by getCurrentTranslation in the generated code cannot be specified by the user
```
Comment (demkos) [inline src/Propel/Generator/Behavior/I18n/I18nBehaviorObjectBuilderModifier.php:221]:
```suggestion
     * @todo: the connection used by getCurrentTranslation in the generated code cannot be specified by the user
```

File: src/Propel/Generator/Behavior/I18n/I18nBehaviorObjectBuilderModifier.php [hunk 6]
```diff
@@ -182,8 +248,13 @@ protected function addTranslatedColumnGetter(Column $column)
...
```
Comment (demkos) [inline src/Propel/Generator/Behavior/I18n/I18nBehaviorObjectBuilderModifier.php:252]:
```suggestion
     * @todo: the connection used by getCurrentTranslation in the generated code cannot be specified by the user
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehavior.php [hunk 7]
```diff
@@ -69,6 +69,9 @@ public function modifyTable()
         }
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\NestedSet\NestedSetBehavior|\Propel\Generator\Behavior\NestedSet\NestedSetBehaviorObjectBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehavior.php:73]:
```suggestion
     * @return \Propel\Generator\Behavior\NestedSet\NestedSetBehaviorObjectBuilderModifier
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehavior.php [hunk 8]
```diff
@@ -78,6 +81,9 @@ public function getObjectBuilderModifier()
         return $this->objectBuilderModifier;
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\NestedSet\NestedSetBehavior|\Propel\Generator\Behavior\NestedSet\NestedSetBehaviorQueryBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehavior.php:85]:
```suggestion
     * @return \Propel\Generator\Behavior\NestedSet\NestedSetBehaviorQueryBuilderModifier
```

File: src/Propel/Runtime/ActiveRecord/NestedSetRecursiveIterator.php (hunk 4 above)
Comment (dereuromark) [reply src/Propel/Runtime/ActiveRecord/NestedSetRecursiveIterator.php:84]: Check php core method docblock 
It says it cannot return and any return is ignored.

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php [hunk 9]
```diff
@@ -21,43 +21,75 @@
  */
 class NestedSetBehaviorObjectBuilderModifier
 {
-    /** @var NestedSetBehavior */
+    /**
+     * @var NestedSetBehavior
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php:25]:
```suggestion
     * @var NestedSetBehavior
```
```suggestion
     * @var \Propel\Generator\Behavior\NestedSet\NestedSetBehavior
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php [hunk 10]
```diff
@@ -21,43 +21,75 @@
...
+     */
     protected $behavior;
 
-    /** @var Table */
+    /**
+     * @var Table
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php:30]:
```suggestion
     * @var \Propel\Generator\Model\Table
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php [hunk 11]
```diff
@@ -21,43 +21,75 @@
...
+     */
     protected $table;
 
-    /** @var ObjectBuilder */
+    /**
+     * @var ObjectBuilder
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorObjectBuilderModifier.php:35]:
```suggestion
     * @var \Propel\Generator\Builder\Om\ObjectBuilder
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php [hunk 12]
```diff
@@ -20,37 +20,70 @@
  */
 class NestedSetBehaviorQueryBuilderModifier
 {
-    /** @var NestedSetBehavior */
+    /**
+     * @var NestedSetBehavior
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php:24]:
```suggestion
     * @var \Propel\Generator\Behavior\NestedSet\NestedSetBehavior
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php [hunk 13]
```diff
@@ -20,37 +20,70 @@
...
+     */
     protected $behavior;
 
-    /** @var Table */
+    /**
+     * @var Table
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php:29]:
```suggestion
     * @var \Propel\Generator\Model\Table
```

File: src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php [hunk 14]
```diff
@@ -20,37 +20,70 @@
...
+     */
     protected $table;
 
-    /** @var QueryBuilder */
+    /**
+     * @var QueryBuilder
```
Comment (demkos) [inline src/Propel/Generator/Behavior/NestedSet/NestedSetBehaviorQueryBuilderModifier.php:34]:
```suggestion
     * @var \Propel\Generator\Builder\Om\QueryBuilder
```

File: src/Propel/Generator/Behavior/Sluggable/SluggableBehavior.php [hunk 15]
```diff
@@ -213,10 +240,13 @@ protected function createRawSlug()
         $script .= "
 }
 ";
-
-        return $script;
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Sluggable/SluggableBehavior.php:217]: Should be reverted?

File: src/Propel/Generator/Behavior/Sortable/SortableBehavior.php [hunk 16]
```diff
@@ -64,6 +64,9 @@ public function modifyTable()
         }
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\Sortable\SortableBehavior|\Propel\Generator\Behavior\Sortable\SortableBehaviorObjectBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Sortable/SortableBehavior.php:68]:
```suggestion
     * @return \Propel\Generator\Behavior\Sortable\SortableBehaviorObjectBuilderModifier
```

File: src/Propel/Generator/Behavior/Sortable/SortableBehavior.php [hunk 17]
```diff
@@ -73,6 +76,9 @@ public function getObjectBuilderModifier()
         return $this->objectBuilderModifier;
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\Sortable\SortableBehavior|\Propel\Generator\Behavior\Sortable\SortableBehaviorQueryBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Sortable/SortableBehavior.php:80]:
```suggestion
     * @return \Propel\Generator\Behavior\Sortable\SortableBehaviorQueryBuilderModifier
```

File: src/Propel/Generator/Behavior/Sortable/SortableBehavior.php [hunk 18]
```diff
@@ -82,6 +88,9 @@ public function getQueryBuilderModifier()
         return $this->queryBuilderModifier;
     }
 
+    /**
+     * @return \Propel\Generator\Behavior\Sortable\SortableBehavior|\Propel\Generator\Behavior\Sortable\SortableBehaviorTableMapBuilderModifier
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Sortable/SortableBehavior.php:92]:
```suggestion
     * @return \Propel\Generator\Behavior\Sortable\SortableBehaviorTableMapBuilderModifier
```

File: src/Propel/Generator/Behavior/Sortable/SortableBehavior.php [hunk 19]
```diff
@@ -174,7 +185,7 @@ public function getColumnSetter($name)
     }
 
     /**
-     * {@inheritdoc}
+     * @inheritdoc
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Sortable/SortableBehavior.php:188]: Revert?

File: src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php [hunk 20]
```diff
@@ -70,6 +75,12 @@ protected function getColumnSetter($column)
         return 'set' . $this->getColumnForParameter($column)->getPhpName();
     }
 
+    /**
+     * @param string $columnName
+     * @param \Propel\Generator\Builder\Om\AbstractOMBuilder $builder
+     *
+     * @return mixed
```
Comment (demkos) [inline src/Propel/Generator/Behavior/Timestampable/TimestampableBehavior.php:82]:
```suggestion
     * @return string
```

File: src/Propel/Generator/Behavior/Sortable/SortableBehavior.php (hunk 19 above)
Comment (dereuromark) [reply src/Propel/Generator/Behavior/Sortable/SortableBehavior.php:188]: Why?

File: src/Propel/Generator/Behavior/I18n/I18nBehavior.php (hunk 3 above)
Comment (dereuromark) [reply src/Propel/Generator/Behavior/I18n/I18nBehavior.php:285]: Are we planning to not use void but $this? Otherwise we can keep it

Comment (dereuromark) [reply src/Propel/Generator/Behavior/I18n/I18nBehavior.php:285]: I will remove it for now, and we can add it in next release.

File: src/Propel/Common/Config/Loader/IniFileLoader.php [hunk 21]
```diff
@@ -47,12 +47,10 @@ public function supports($resource, $type = null)
     /**
      * Loads a resource, merge it with the default configuration array and resolve its parameters.
      *
-     * @param  mixed  $file The resource
-     * @param  string $type The resource type
+     * @param  string  $file The resource
+     * @param  string|null $type The resource type
      * @return array  The configuration array
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply src/Propel/Common/Config/Loader/IniFileLoader.php:52]: With the below changes it now cannot be null anymore.

**Pull request merged**